ocaml/xcp-rrdd/bin/rrdd/xcp_rrdd.ml

```diff
@@ -361,9 +361,11 @@ let dss_loadavg () =
     )
   ]
 
-let count_running_domain domains =
+let count_power_state_running_domains domains =
   List.fold_left
-    (fun count (dom, _, _) -> if dom.Xenctrl.running then count + 1 else count)
+    (fun count (dom, _, _) ->
+      if dom.Xenctrl.running || dom.Xenctrl.blocked then count + 1 else count
+    )
     0 domains
 
```

Comment:
This is still an incorrect calculation.  It will miscount when the host is oversubscribed with CPU bound workloads.

I already have you the right condition to check.  You need to count the VMs which don't have a shutdown state.

Reply:
Would the VMs with paused flag be counted as `running` as well?

Reply:
There is a dying flag as well, so domain with dying state should not counted as well I.M.O.

Reply:
As test, power-status is running and it keeps a long period during VM suspend and particularly take snapshot for disks and memory. 
How about just treat `not paused` as running?

Reply:
This is snapshot test, `ss` and `running` status last about 90 seconds.
![image](https://github.com/xapi-project/xen-api/assets/26405345/9b93a30c-4959-4dcc-84ef-3d63a13a8cae)

Reply:
At the same time, the VM was flagged with a yellow indicator in XenCenter, so it makes sense not to count it as `running`

Reply:
Given we focus on the power-state running domain, I updated the condition to exclude the paused state domain with this metrics.